completion/src/org/jetbrains/kotlin/idea/completion/KeywordCompletion.kt

```diff
@@ -439,6 +439,10 @@ object KeywordCompletion {
                         elementAt.parent is KtParameter &&
                         elementAt.parentOfTypes(KtNamedFunction::class, KtSecondaryConstructor::class) != null -> return false
 
+                keywordTokenType == CONSTRUCTOR_KEYWORD && elementAt.parent.parent.getStrictParentOfType<KtDeclaration>()?.let {
```

Comment:
I would suggest to encapsulate the logic into a local fun like this:
```
fun PsiElement.isInObjectDeclaration(): Boolean {
  val secondaryConstructor = parentOfType<KtSecondaryConstructor>() ?: return false
  secondaryConstructor.getContainingClassOrObject() is KtObjectDeclaration
}
```
So that the resulting expression would be:
```
keywordTokenType == CONSTRUCTOR_KEYWORD && elementAt.isInObjectDeclaration()
```

Reply:
> I would suggest to encapsulate the logic into a local fun like this:

Done